src/components/input/icon-picker/IconPicker.tsx

```diff
@@ -0,0 +1,125 @@
+import classes from './IconPicker.module.scss';
+
+import { useCallback, useEffect, useState } from 'react';
+import { IconKeyType, getIconComponent, iconKeys } from '../../../assets/icons';
+import ModalPopup, { ButtonType } from '../../generic/ModalPopup';
+import TextField from '../text-field/TextField';
+import classNames from 'classnames/bind';
+import { useDebouce } from '../../../hooks/useDebounce';
+import { deepEqual } from '../../../utils/comparison';
+
+const cx = classNames.bind(classes);
+
+interface IconPickerProps {
+    className?: string;
+    label: string;
+    iconKey: IconKeyType;
+    onIconChange: (iconKey: IconKeyType) => void;
+}
+
+const filterIconKeys = (query: string, iconKeys: IconKeyType[]) => {
+    const cleanedQuery = query.toLowerCase().trim();
+    return iconKeys.filter((iconKey) =>
+        iconKey.toLowerCase().includes(cleanedQuery)
+    );
+};
+
+const IconPicker = ({
+    className,
+    label,
+    iconKey,
+    onIconChange,
+}: IconPickerProps) => {
+    const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+    const [searchQuery, setSearchQuery] = useState<string>('');
+    const debouncedSearchQuery = useDebouce(searchQuery);
+    const [selectedIconKey, setSelectedIconKey] =
+        useState<IconKeyType>(iconKey);
+    const [displayedIconKeys, setDisplayedIconKeys] =
+        useState<IconKeyType[]>(iconKeys);
+
+    useEffect(() => {
+        debouncedSearchQuery
+            ? setDisplayedIconKeys(
+                  filterIconKeys(debouncedSearchQuery, iconKeys)
+              )
+            : setDisplayedIconKeys(iconKeys);
+    }, [debouncedSearchQuery]);
+
+    const sameIcon = deepEqual(iconKey, selectedIconKey);
```

Comment:
`deepEqual` is a heavy operation that shouldn't be called without a `useMemo` caching its result